presto-main/src/main/java/com/facebook/presto/sql/planner/StatisticsAggregationPlanner.java

```diff
@@ -98,18 +112,63 @@ public TableStatisticAggregation createStatisticsAggregation(TableStatisticsMeta
             VariableReferenceExpression inputVariable = columnToVariableMap.get(columnName);
             verify(inputVariable != null, "inputVariable is null");
             ColumnStatisticsAggregation aggregation = createColumnAggregation(columnStatisticMetadata, inputVariable);
+            additionalVariables.putAll(aggregation.getInputProjections());
             VariableReferenceExpression variable = variableAllocator.newVariable(statisticType + ":" + columnName, aggregation.getOutputType());
             aggregations.put(variable, aggregation.getAggregation());
             descriptor.addColumnStatistic(columnStatisticMetadata, variable);
         }
 
         StatisticAggregations aggregation = new StatisticAggregations(aggregations.build(), groupingVariables);
-        return new TableStatisticAggregation(aggregation, descriptor.build());
+        return new TableStatisticAggregation(aggregation, descriptor.build(), additionalVariables.build());
     }
 
-    private ColumnStatisticsAggregation createColumnAggregation(ColumnStatisticMetadata columnStatisticMetadata, VariableReferenceExpression input)
+    private ColumnStatisticsAggregation createColumnAggregationFromSqlFunction(String sqlFunction, VariableReferenceExpression input)
     {
-        FunctionHandle functionHandle = functionAndTypeResolver.lookupFunction(columnStatisticMetadata.getFunctionName(), TypeSignatureProvider.fromTypes(ImmutableList.of(input.getType())));
+        RowExpression expression = sqlFunctionToRowExpression(
+                sqlFunction,
+                ImmutableSet.of(input),
+                functionAndTypeManager,
+                session);
+        verify(expression instanceof CallExpression, "column statistic SQL expressions must represent a function call");
+        CallExpression call = (CallExpression) expression;
+        verify(functionAndTypeResolver.getFunctionMetadata(call.getFunctionHandle())
+                        .getFunctionKind().equals(AGGREGATE),
```

Comment:
nit: Extract `functionMetadata` as a variable to avoid multiple calls of `getFunctionMetadata(...)`.